### PR TITLE
DOC: update pandas.Series.rename_axis

### PR DESCRIPTION
**Pull request by xrmx**
Checklist for the pandas documentation sprint (ignore this if you are doing
an unrelated PR):

- [X] PR title is "DOC: update the <your-function-or-method> docstring"
- [X] The validation script passes: `scripts/validate_docstrings.py <your-function-or-method>`
- [X] The PEP8 style check passes: `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [X] The html version looks good: `python doc/make.py --single <your-function-or-method>`
- [X] It has been proofread on language by another sprint participant

Please include the output of the validation script below between the "```" ticks:

```
################################################################################
############# Docstring (pandas.core.generic.NDFrame.rename_axis)  #############
################################################################################

Alter the name of the index or columns.

This function alters the name of the index or columns.

Parameters
----------
mapper : scalar, list-like, optional
    Value to set the axis name attribute.
axis : int, str, default 0
    The index or the name of the axis.
copy : boolean, default True
    Also copy underlying data.
inplace : boolean, default False
    Modifies the mapper in place.

Returns
-------
type of caller or None if inplace=True
    Renamed

Notes
-----
Prior to version 0.21.0, ``rename_axis`` could also be used to change
the axis *labels* by passing a mapping or scalar. This behavior is
deprecated and will be removed in a future version. Use ``rename``
instead.

See Also
--------
pandas.Series.rename : Alter Series index labels or name
pandas.DataFrame.rename : Alter DataFrame index labels or name
pandas.Index.rename : Set new names on index

Examples
--------
>>> df = pd.DataFrame({"A": [1, 2, 3], "B": [4, 5, 6]})
>>> df.rename_axis("foo")
     A  B
foo
0    1  4
1    2  5
2    3  6

>>> df.rename_axis("bar", axis="columns")
bar  A  B
0    1  4
1    2  5
2    3  6

################################################################################
################################## Validation ##################################
################################################################################

Docstring for "pandas.core.generic.NDFrame.rename_axis" correct. :)
```

### CODE REVIEW

File: pandas/core/generic.py [hunk 1]
```diff
@@ -913,20 +913,26 @@ def f(x):
     rename.__doc__ = _shared_docs['rename']
 
     def rename_axis(self, mapper, axis=0, copy=True, inplace=False):
-        """Alter the name of the index or columns.
+        """
+        Alter the name of the index or columns.
+
+        This function alters the name of the index or columns.
```
Comment (TomAugspurger) [inline pandas/core/generic.py]: Since this is just restating the short summary I think it can be removed.

File: pandas/core/generic.py [hunk 2]
```diff
@@ -913,20 +913,26 @@ def f(x):
...
 
         Parameters
         ----------
         mapper : scalar, list-like, optional
             Value to set the axis name attribute.
-        axis : int or string, default 0
+        axis : int, str, default 0
+            The index or the name of the axis.
```
Comment (TomAugspurger) [inline pandas/core/generic.py]: Where inconsistent with axis. But

            axis : {0 or 'index', 1 or 'columns'}, default 0
seems to be the most common.

File: pandas/core/generic.py [hunk 3]
```diff
@@ -913,20 +913,26 @@ def f(x):
...
         copy : boolean, default True
-            Also copy underlying data
+            Also copy underlying data.
         inplace : boolean, default False
+            Modifies the mapper in place.
 
         Returns
         -------
-        renamed : type of caller or None if inplace=True
+        type of caller or None if inplace=True
```
Comment (TomAugspurger) [inline pandas/core/generic.py]: I think this should be

    renamed : Series, DataFrame, or None
        The same type as the caller or None if `inplace` is True.

File: pandas/core/generic.py (hunk 1 above)
Comment (xrmx) [reply pandas/core/generic.py]: Done.

File: pandas/core/generic.py (hunk 2 above)
Comment (xrmx) [reply pandas/core/generic.py]: Done

File: pandas/core/generic.py (hunk 3 above)
Comment (xrmx) [reply pandas/core/generic.py]: Done

**Comment by xrmx** [comment]
Updated PR after review, thanks!

**Comment by TomAugspurger** [comment]
Added a series example. Thanks @xrmx!

**Pull request merged**